core-models/src/main/scala/com/mesosphere/usi/core/models/StateEvent.scala

```diff
@@ -27,6 +27,7 @@ object StateSnapshot {
   */
 sealed trait StateUpdated extends StateEvent
 
+case class PodInvalid(id: PodId, errors: Seq[String]) extends StateUpdated with PodStateEvent
```

Comment:
After thinking about this more... it really is a bug if the framework provides invalid pod specs. I don't think we should have a special status message for it. Can we just use podStatus? Expose the message the same way we would a regular failure.

Reply:
I don't want to encode that in `case class PodStatus(id: PodId, taskStatuses: Map[TaskId, Mesos.TaskStatus])`. Maybe we can derive another type from PodStatus with extra fields and empty task statuses? WDYT, @timcharper

Reply:
I agree with Alena, having an empty `tasksStatuses` and an `errors: Seq[String]` there seems weird. I'd rather make the API expressive and readable. 
But maybe we want to a general type for errors? Not every error will be Pod related e.g. we might need to bubble up Mesos `message Error` event.

Reply:
Oh man. This would be super hard to deal with from the framework perspective. Please help me understand. Is there any validation that we cannot do during the `PodSpecUpdated` construction?

Reply:
Why is it a `StateUpdated`? Should it not extend a `ErrorEvent` or something similar?

Reply:
Ping @alenkacz